src/main/scala/strawman/collection/immutable/HashMap.scala

```diff
@@ -104,6 +100,12 @@ object HashMap extends MapFactory[HashMap] {
 
   def empty[K, V]: HashMap[K, V] = EmptyHashMap.asInstanceOf[HashMap[K, V]]
 
+  def fromIterable[K, V](it: collection.Iterable[(K, V)]): HashMap[K, V] =
+    it match {
+      case hm: HashMap[K, V] => hm
+      case _ => empty ++ it
```

Comment:
For an immutable array, for example, relying on `++` requires this to be optimized for corner cases like the LHS being empty. Maybe that's something we'd want to have anyway but when using a Builder you already know this precondition.